src/messaging/index.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * KILT participants can communicate via a 1:1 messaging system.
+ *
+ * All messages are **encrypted** with the encryption keys of the involved identities.
+ * Messages are encrypted using authenticated encryption: the two parties authenticate to each other, but the message authentication provides repudiation possibilities.
+ *
+ * The [[Message]] class exposes methods to construct and verify messages.
+ *
+ * @module @kiltprotocol/messaging
+ */
+
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+export * from './utils'
+export * from './MessageEnvelope'
+export * from './CredentialApiMessageType'
```

Comment:
I'm thinking the only things that need to be available on the public interface of the package are:
- encrypt
- decrypt
- fromBody
- assertKnownMessage
- maybe the Errors?

Reduces clutter, avoids overwhelming the users.
@weichweich ?

Reply:
less: [0a2ffb9](https://github.com/KILTprotocol/kilt-extension-api/pull/18/commits/0a2ffb99c122c17f358bd9701cf0754cbc2e8027)